rasa/utils/tf_model_data.py

```diff
@@ -0,0 +1,535 @@
+import numpy as np
+import scipy.sparse
+import tensorflow as tf
+
+from sklearn.model_selection import train_test_split
+from typing import Optional, Dict, Text, List, Tuple, Any, Union, Generator, NamedTuple
+from collections import defaultdict
+
+
+class DataSignature(NamedTuple):
```

Comment:
there is a discrepancy in names: we use `data_signature` to name the dict of `DataSignature` objects. Should `DataSignature` be a dictionary?

Reply:
What about we name the named tuple `FeatureSignature` instead of `DataSignature`?

Reply:
ok